includes/classes/Feature/Search/Synonyms.php

```diff
@@ -187,7 +187,7 @@ public function admin_notices() {
 			return;
 		}
 
-		$update = filter_input( INPUT_GET, 'ep_synonym_update', FILTER_SANITIZE_STRING );
+		$update = filter_input( INPUT_GET, 'ep_synonym_update', FILTER_SANITIZE_SPECIAL_CHARS );
```

Comment:
This does not strip tags, but the change should be safe as we only compare the value with the predefined set of constants.